File: Baekjoon/Lv.6/1065.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        System.out.print(arithmeticSequence(Integer.parseInt(br.readLine())));
    }

    public static int arithmeticSequence(int num) {
        int count = 0;

        if (num < 100) {
            return num;
        } else {
            count = 99;
            if (num == 1000) { // 1000의 경우, 예외처리
                num = 999;
            }

            for (int i = 100; i <= num; i++) {
                int hun = i / 100; // 백의 자리
                int ten = (i / 10) % 10; // 십의 자리
                int one = i % 10;

                if ((hun - ten) == (ten - one)) { // 각 자릿수가 수열을 이루면
                    count++;
                }
            }
        }

        return count;
    }
}

File: Study Member/Newon/문제/1주차 문제/예제문제/level-6/1065.kt
import java.io.*

fun main(){
    val br = System.`in`.bufferedReader()
    
    val n = br.readLine().toInt()
    var i = 100
    var han = 99
    var arr = Array(3,{0})
    
    if (n < 100){
        println(n)
    } else if (n < 1000){
        while (i <= n) {
            arr[0] = i / 100;
	        arr[1] = (i / 10) % 10;
	        arr[2] = i % 10;
	        if (arr[0] - arr[1] == arr[1] - arr[2]){
		    han++;
            }
	    i++;
        }
    println(han)
    } else if (n == 1000) {
        println("144")
    }
}
